### PR TITLE
Migrate to keycloak

### PR DESCRIPTION
**Pull request by Adzz**
in order to add `ueberauth_oidcc` I had to update `telemetry_poller` which triggered updating phoenix.

This then migrates to using keycloak and updates the tests for that use case.

### CODE REVIEW

**Comment by napilla** [comment]
@Adzz  I made some changes here, mostly to fix some warnings that came up failing the build and test step and making sure it integrates well with Keycloak DEV (looks like it worked well for me). Last steps here are to fix up the tests happy to work on that today

**Comment by Adzz** [comment]
Hey no worry the tests are a nightmare so I can crack that out now, thanks for the contribution though!

**Comment by napilla** [comment]
I'm still reviewing some other stuff in this PR, but it looks like its failing and I have one important thing that I think will need to be done to successfully work in deployed environments: can you add in `ca-certificates` as a dependency in this project's dockerfile? 

See:
https://github.com/mbta/alerts_concierge/blob/d7b0b431bb2e25864e008bb17dbaaa3367387bf6/Dockerfile#L59 

And the error that happens if you do not do this: https://www.notion.so/mbta-downtown-crossing/Implementing-a-Keycloak-integration-ebf669208d464986b4989294891b1d8e?pvs=4#49283e6e8bc0478e965a416f8cc5e8d8

This happened while I was testing APA and suspect it would happen here too.

File: lib/document_viewer/application.ex [hunk 1]
```diff
@@ -11,7 +11,8 @@ defmodule DocumentViewer.Application do
       DocumentViewerWeb.Telemetry,
       # Start the PubSub system
       {Phoenix.PubSub, name: DocumentViewer.PubSub},
-      Catalog.Supervisor,
+      # commenting out so I can start the app or attempt to.
+      # Catalog.Supervisor,
```
Comment (napilla) [inline lib/document_viewer/application.ex:15]: we need this enabled when we merge in

File: test/document_viewer_web/controllers/auth_controller_test.exs [hunk 2]
```diff
@@ -56,23 +54,18 @@ defmodule DocumentViewerWeb.AuthControllerTest do
...
-          |> Plug.Test.init_test_session(%{username: "test@mbta.com"})
-          |> assign(:ueberauth_failure, %Ueberauth.Failure{
-            errors: [%Ueberauth.Failure.Error{message: "failed"}]
-          })
-          |> get("/auth/cognito/callback")
+          conn |> get(~p"/auth/keycloak/callback")
         end)
 
-      assert log =~ "Ueberauth error: failed"
+      assert log =~ "Ueberauth error: Cross-Site Request Forgery attack"
```
Comment (napilla) [inline test/document_viewer_web/controllers/auth_controller_test.exs:68]: hahah, I don't like it but I was having trouble with this yesterday anyhow.

**Review by napilla — COMMENTED** [review]
Looks good, I think we have to enable the one job before we merge though

File: lib/document_viewer/application.ex (hunk 1 above)
Comment (Adzz) [reply lib/document_viewer/application.ex:15]: oh yes great spot! will change

File: test/document_viewer_web/controllers/auth_controller_test.exs (hunk 2 above)
Comment (Adzz) [reply test/document_viewer_web/controllers/auth_controller_test.exs:68]: I think ultimately we are trying to test what happens when we get an error in ueberauth, which this does so I'm fine with it.

**Review by napilla — APPROVED** [review]

**Pull request merged**